PythonVisualizations/BinaryTreeBase.py

```diff
@@ -0,0 +1,515 @@
+   def createHighlightedValue(self, node, color="green"):
+      return self.canvas.create_text(node.coords[0], node.coords[1], text=node.getKey(), font=self.VALUE_FONT,
+                                               fill=color)
+
+   # highlight the value associated with the node
+   def highlightValue(self, node, color="red"):
+      self.canvas.itemconfig(node.drawable.display_val, fill=color)
+
+   # unhighlight the value associated with the node
+   def unHighlightValue(self, node):
+      self.canvas.itemconfig(node.drawable.display_val, fill="black")
+
+   # draws the circle and the key value
+   def createNodeShape(self, x, y, key, tag):
+      circle = self.canvas.create_circle(x, y, self.CIRCLE_SIZE, tag = tag)
+      circle_text = self.canvas.create_text(x,y, text=key, font=self.VALUE_FONT, tag = tag)
+
+      # get rid of the none shape that was in its spot
+      items = set(self.canvas.find_overlapping(*(self.canvas.coords(circle))))
+      noneShape = set(self.canvas.find_withtag("none"))
+      noneItems = (items & noneShape)
+      
+      while len(noneItems) > 0:
+         self.canvas.delete(noneItems.pop())
+
+      self.canvas.update()
+
+      return circle, circle_text
+
+   # creates the none shape that nodes will point to
+   # line coords should be the coords of the line that points to none
+   def createNoneShape(self, node, level, lineCoords, direction, tag):
+      y = lineCoords[3] + (self.CIRCLE_SIZE // 2)
+      xCoord = self.calculateCoordinates(node, level+1, direction)[0]
+
+      shrinkFactor = self.CIRCLE_SIZE // 3
+
+      coords1 = ( xCoord-self.CIRCLE_SIZE, y, 
+                  xCoord+self.CIRCLE_SIZE, y)
+      coords2 = (  shrinkFactor + coords1[0], coords1[1] + shrinkFactor,
+                  -shrinkFactor + coords1[2], coords1[3] + shrinkFactor)
+      coords3 = (  shrinkFactor + coords2[0], coords2[1] + shrinkFactor, 
+                  -shrinkFactor + coords2[2], coords2[3] + shrinkFactor)
+
+      self.canvas.create_line(coords1, tag = (tag, "none"))
+      self.canvas.create_line(coords2, tag = (tag, "none"))
+      self.canvas.create_line(coords3, tag = (tag, "none"))
+
+   def removeNoneShapes(self, node):
+      # get the none shapes associated with the node
+      nodeShapes = set(self.canvas.find_withtag(node.tag))
+      noneShape = set(self.canvas.find_withtag("none"))
+      noneItems = (nodeShapes & noneShape)
+
+      while len(noneItems) > 0:
+         self.canvas.delete(noneItems.pop())
+
+   # delete the node's drawing
+   def removeNodeDrawing(self, node):
+      leftChild = self.getLeftChild(node)
+      rightChild =  self.getRightChild(node)
+
+      self.canvas.delete(node.drawable.display_val)
+      self.canvas.delete(node.drawable.display_shape)
+      self.canvas.delete(self.getRightLine(node))
+      self.canvas.delete(self.getLeftLine(node))
+      self.removeNoneShapes(node)
+
+   def moveNodesToPos(self, moveItems):
+      # get the coords for the node to move to
```

Comment:
Like the other animations, I think this should have `steps` and `sleepTime` parameters to control the movement.  I also think the name something like `restoreNodesPosition` since it moves the items to their defined location in the tree.  With the name `moveNodesToPos`, it makes me expect another parameter for the position, and the ability to move nodes to any position.

Reply:
Changes implemented in commit bcab57877738ecfd6c39c5c88c77c17dbb667869